HIP/hip-0000-nft-standard.md

```diff
@@ -0,0 +1,795 @@
+
+References: https://w3c.github.io/webappsec-subresource-integrity/
+
+
+### files.type
+
+**Type:** string (MIME type)
+
+**Required**
+
+**Description:** Mime formatting shall follow the following format: **type/subtype**. As a rule, mime types are all lower case. However apps should be programmed to accept any case for robustness.
+
+A list of common mime types can be found here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+
+Note that mime types for directories are not uniformly defined. Some IPFS CIDs point to directories rather than files, so this type is useful. This standard shall define it using the format: text/directory.
+
+
+### files.is_default_file
+
+**Type:** boolean (false/true)
+
+**Conditionally optional:** Required when multiple `file` objects are present in the `files` array. When only one `file` object is defined, this becomes the default file.
+
+**Description:** The `is_default_file` property makes it easier for NFT tooling to identify the main file for an NFT if multiple files are present.
+
+
+### files.metadata
+
+**Type:** object
+
+**Optional**
+
+**Description:**  This is a nested metadata object for the file, which follows the same metadata format as the root metadata. Files can be nested indefinitely in this way, but processed with the same metadata code.
+
+
+### files.metadata_uri
+
+**Type:** string (CID or URI)
+
+**Optional**
+
+**Description:** There are situations (such as mutable metadata) where rather than including the "metadata" object it makes sense to point to a different file. Therefore this is a URI that points to a metadata json file for the file in question. 
+
+**To avoid conflicts, if "metadata" is defined then "metadata_uri" should be ignored.**
+
+An NFT creator has the option of using either "metadata" or "metadata_uri". Metadata should be considered the default behaviour as it minimizes the number of calls that need to be made. Metadata_uri should be used in specific situations where defining the metadata object within the base metadata file is inadequate.
+
+
+### attributes
+
+**Type:** array
+
+**Required**
+
+**Description:** Required array of attribute objects to calculate the rarity score of NFTs. If you don't wish to implement rarity scores, add an empty array. 
+
+
+### attributes.trait_type
+
+**Type:** string
+
+**Required**
+
+**Description:** Name of trait.
+
+
+### attributes.display_type
+
+**Type:** string - Possible display types (but other types are allowed):
+
+- `text` (**default value representation**)
+- `percentage` (for integer or number values)
+- `boost` (for integer or number values)
+- `datetime` (for string values)
```

Comment:
Opensea have a good example of the `date` display_type. We can extend this to support `datetime` as well to include the time. I've added color as well to give some inspiration to how that could work.

https://docs.opensea.io/docs/metadata-standards#date-traits

```suggestion
- `datetime` (for a number which represents the unix timestamp in seconds )
- `date` (for a number which represents the unix timestamp in seconds )
- `color` (for a hexadecimal or rgb string color sequence such as "#00ff44" or "rgb(0,255,0)" )
```

Reply:
I like this!